viewer/packages/tools/src/Timeline/TimelineTool.test.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import { Cognite3DModel } from '@reveal/core';
+import { CogniteClient } from '@cognite/sdk';
+import { CadMaterialManager, CadNode } from '@reveal/rendering';
+import { CadModelMetadata } from '@reveal/cad-parsers';
+import { createCadModelMetadata, generateSectorTree } from '../../../../test-utilities';
+import { NodesLocalClient } from '@reveal/nodes-api';
+
+import { IndexSet } from '../../../../packages/utilities';
+import { TreeIndexNodeCollection } from '../../../../packages/cad-styling';
+import nock from 'nock';
+import { TimelineTool } from './TimelineTool';
+import TWEEN from '@tweenjs/tween.js';
+
+describe('TimelineTool', () => {
+  let client: CogniteClient;
+  let model: Cognite3DModel;
+
+  beforeEach(() => {
+    client = new CogniteClient({ appId: 'test', baseUrl: 'http://localhost' });
+    client.loginWithApiKey({ apiKey: 'dummy', project: 'unittest' });
+
+    const materialManager = new CadMaterialManager();
+    const cadRoot = generateSectorTree(3, 3);
+    const cadMetadata: CadModelMetadata = createCadModelMetadata(cadRoot);
+    materialManager.addModelMaterials(cadMetadata.modelIdentifier, cadMetadata.scene.maxTreeIndex);
+
+    const cadNode = new CadNode(cadMetadata, materialManager);
+    const apiClient = new NodesLocalClient();
+
+    model = new Cognite3DModel(1, 2, cadNode, apiClient);
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    nock.cleanAll();
+  });
+
+  test('Test keyframe creation', () => {
+    const timelineTool = new TimelineTool(model);
+    expect(timelineTool).toBeTruthy();
+
+    const kf1 = timelineTool.createKeyframe(new Date('2021-10-25'));
+    const kf2 = timelineTool.createKeyframe(new Date('2021-10-26'));
+    const kf3 = timelineTool.createKeyframe(new Date('2021-10-27'));
+    const kf4 = timelineTool.createKeyframe(new Date('2021-10-28'));
+
+    expect(kf1).not.toBeEmpty();
+    expect(kf2).not.toBeEmpty();
+    expect(kf3).not.toBeEmpty();
+    expect(kf4).not.toBeEmpty();
```

Comment:
`toBeEmpty` works on collections and strings, not objects. Use `not.toBeNullOrUndefined()`.

Not sure this test makes sense. Adding many keyframes doesn't add a lot to the test. A more interesting test would be to add e.g. three keyframes with shuffled dates and ensure they are ordered correctly in `getKeyframes()`.